server/__tests__/server.test.js

```diff
@@ -0,0 +1,49 @@
+import { Mockgoose } from 'mockgoose';
+import mongoose from 'mongoose';
+import server from '../../server';
+import User from '../models/user';
+
+// const request = require('supertest');
+const session = require('supertest-session');
+const verifiedUser = require('../__mockData__/verifiedUser.json');
+
+const mockgoose = new Mockgoose(mongoose);
+
+var testSession = session(server, {
+  before: function (req) {
+    req.set('user', {username: 'no-stack-dub-sack'});
+  }
+});
+
+describe('api endpoints', () => {
+  beforeAll(() => {
+    // mockgoose intercepts the connection, no connection is made
+    mockgoose.prepareStorage().then(function() {
+      mongoose.connect('mongodb://localhost/testing-db', { useMongoClient: true }, err => {
```

Comment:
I think that once I can get around the setting `user` on the req object, that this `beforeAll` hook should hijack the mongo connection and when `User.findOne(...)` is called in the route, it should instead use the mocked version of mongoose and find the data that I've inserted in the `before` hook below. Pulling it in from the JSON file in `__mockData__` and know this works because the first test passes. Though that is calling `User.findOne(...)` directly in this scope, I'm still a little unclear on whether or not mockgoose will successfully stub out the mongo call in the route.